pkg/kubernetes/event_handlers.go

```diff
@@ -68,6 +68,9 @@ func GetKubernetesEventHandlers(informerName, providerName string, announce chan
 				logNotObservedNamespace(newObj, eventTypes.Update)
 				return
 			}
+			if reflect.DeepEqual(oldObj, newObj) {
+				return
+			}
```

Comment:
This does remove the eventual consistency mechanism. Would be good to call it out in the PR description and code.

Reply:
@shashankram I am not quite certain what you are referring to, but I believe that change would not have any adverse affects on pushing config to Envoys.
I'm in no hurry to push this.  I know you had worked on this in the past - we can leave the PR to remind us that this here is an optimization we should add.

Reply:
This is definitely a good think to have. From my understanding, this check was removed in the past to adopt an eventual consistency model where regardless of `newObj != oldObj`, the updates were let through. @eduser25 is focusing on filtering announcements, I'll let him clarify if he has thoughts on this.

Reply:
agreed. 
@Del, we just figured out there are missing pieces for this to even work. Even though in paper this should be ok, I think there's some notifications (ie. config) not guaranteed to be handled right now in certain circumstances, hence the resync is acting as the overall mechanism to make sure state is always up to date.

Reply:
@eduser25 I'll keep this PR open.  I want to understand what these circumstances are. It seems to me that we should have a consistent system without relying on resyncs.